jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppClassLoader.java

```diff
@@ -282,11 +283,19 @@ public void addClassPath(String classPath)
 
     /**
      * @param file Checks if this file type can be added to the classpath.
+     * TODO: move to FileID in later PR
      */
+
     private boolean isFileSupported(String file)
     {
         int dot = file.lastIndexOf('.');
-        return dot != -1 && _extensions.contains(file.substring(dot));
+        return dot != -1 && _extensions.contains(file.substring(dot).toLowerCase(Locale.ENGLISH));
+    }
+
+    // TODO: move to FileID in later PR
+    private boolean isFileSupported(Path path)
+    {
+        return isFileSupported(path.getFileName().toString());
```

Comment:
Did you know that `Path.of("/").getFileName()` returns `null`?

Reply:
I'm aware of that.  
This is why the use of this is in a filter preceded by `.filter(Files:isRegularFile)`